sdk/test-utils/recorder-new/MIGRATION.md

```diff
@@ -0,0 +1,160 @@
+| `unit-test:browser` | `karma start --single-run`                                                                                                                                                               | `dev-tool run test:browser`                                                                                                            |
+| `unit-test:node`    | `mocha -r esm -r ts-node/register --reporter ../../../common/tools/mocha-multi-reporter.js --timeout 1200000 --full-trace --exclude \"test/**/browser/*.spec.ts\" \"test/**/*.spec.ts\"` | `dev-tool run test:node-ts-input --mocha=\"--timeout 1200000 --full-trace --exclude 'test/**/browser/*.spec.ts' 'test/**/*.spec.ts'\"` |
+
+## Initializing the recorder
+
+The approach taken to initialize the recorder depends on whether the SDK being tested uses Core v1 (`core-http`) or Core v2 (`core-rest-pipeline`). If your SDK is on Core v2, read on. If you're still on Core v1, jump to the section on Core v1 below.
+
+### For Core v2 SDKs
+
+The recorder is implemented as a custom policy which should be attached to your client's pipeline. Firstly, initialize the recorder:
+
+```ts
+let recorder: TestProxyHttpClient;
+
+/*
+ * Note the use of function() instead of the arrow syntax. We need access to `this` so we
+ * can pass test information from Mocha to the recorder.
+ */
+beforeEach(function(this: Context) {
+  recorder = new TestProxyHttpClient(this.currentTest);
+});
+```
+
+To enable the recorder, you should then initialize your SDK client as normal and use the recorder's `configureClient` method. This method will attach the necessary policies to the client for recording to be enabled.
+
+```ts
+const client = /* ... initialize your client as normal ... */;
+// recorderHttpPolicy is provided as an export from the test-recorder-new package.
+recorder.configureClient(client);
+```
+
+### For Core v1 SDKs
+
+The recorder library provides a custom `HttpClient` that is then passed to the SDK. This client needs to be initialized as follows:
+
+```ts
+let recorder: TestProxyHttpClientCoreV1;
+
+/*
+ * Note the use of function() instead of the arrow syntax. We need access to `this` so we
+ * can pass test information from Mocha to the recorder.
+ */
+beforeEach(function(this: Context) {
+  recorder = new TestProxyHttpClientCoreV1(this.currentTest);
+});
+```
+
+When initialising your client in your test, you should pass in the recorder as follows:
+
+```ts
+const client = new MyServiceClient(/* ... insert options here ... */, { httpClient: recorder });
+```
+
+This will allow requests to be intercepted and redirected to the proxy tool.
+
+## Starting and stopping the recorder
+
+The recorder must now be started and stopped explicitly. At the beginning of your test, start the recorder as follows:
```

Comment:
Start and stop used to be explicit as well, it's just that `record` has changed to `recorder.start`.